tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java

```diff
@@ -26,22 +41,111 @@ public void run() {
+                getView(request.getEndpoint(), response);
+                return;
+            }
+            String endpoint = request.getEndpoint();
+            Http11Method method = request.getMethod();
+            if (method == Http11Method.GET && "/".equals(endpoint)) {
+                response.addBody("Hello world!");
+                return;
+            }
+            if (method == Http11Method.GET && "/login".equals(endpoint)) {
+                if (hasUser(request)) {
+                    response.sendRedirect("/index.html");
+                    return;
+                }
+                getView("/login.html", response);
+                return;
+            }
+            if (method == Http11Method.GET && "/register".equals(endpoint)) {
+                getView("/register.html", response);
+                return;
+            }
+            if (method == Http11Method.POST && "/login".equals(endpoint)) {
+                login(request, response);
+                return;
+            }
+            if (method == Http11Method.POST && "/register".equals(endpoint)) {
+                register(request, response);
+                return;
+            }
+            response.sendRedirect("/404.html");
+        } catch (Exception e) {
+            response.sendRedirect("/500.html");
+        }
+    }
+
+    private boolean hasUser(Http11Request request) {
+        try {
+            Session session = request.getSession();
+            session.getAttribute("user");
+            return true;
+        } catch (IllegalArgumentException e) {
+        }
+        return false;
+    }
+
+    private void getView(String view, Http11Response response) throws IOException {
+        URL resource = getClass().getClassLoader().getResource("static" + view);
+        if (resource == null) {
+            throw new IllegalArgumentException("존재하지 않는 자원입니다.");
+        }
+        Path path = new File(resource.getFile()).toPath();
+        String contentType = Files.probeContentType(path);
+        response.addBody(new String(Files.readAllBytes(path)));
+        response.addContentType(contentType);
+    }
+
+    private void login(Http11Request request, Http11Response response) {
+        String body = request.getBody();
+        Map<String, List<String>> queryStrings = QueryStringParser.parseQueryString(body);
+        String account = queryStrings.get("account").getFirst();
+        String password = queryStrings.get("password").getFirst();
+
+        Optional<User> optionalUser = InMemoryUserRepository.findByAccount(account);
+        if (optionalUser.isPresent()) {
+            User user = optionalUser.get();
+            if (user.checkPassword(password)) {
+                Session session = request.getSession();
+                session.setAttribute("user", user);
+                response.addCookie("JSESSIONID", session.getId());
+                response.sendRedirect("/index.html");
+                return;
+            }
+        }
```

Comment:
메소드 indent를 줄일 수 있지 않을까요? `Optional<User>`를 활용할 거라면, `ifPresent()` 메소드는 어떨까요?

```suggestion
        InMemoryUserRepository.findByAccount(account).ifPresent((user) -> {
            if (user.checkPassword(password)) {
                Session session = request.getSession();
                session.setAttribute("user", user);
                response.addCookie("JSESSIONID", session.getId());
                response.sendRedirect("/index.html");
                return;
            }
            response.sendRedirect("/401.html");
        });
```

Reply:
오 생각하지 못한 방법인데 반영했습니다.

Reply:
다시 생각해보니 ID가 틀린 경우에 401로 리다이렉트가 되지 않아 롤백했습니다.